src/Networking/NexusMods.Networking.Downloaders/Tasks/ADownloadTask.cs

```diff
@@ -236,6 +245,17 @@ private void UpdateActivity()
             Logger.LogError(ex, "Failed to update activity status");
         }
     }
+    
+    /// <inheritdoc />
+    public async Task SetIsHidden(bool isHidden, ITransaction? tx = null)
+    {
+        if (PersistentState.Status != DownloadTaskStatus.Completed)
+            return;
+        var transaction = tx ?? Connection.BeginTransaction();
+        transaction.Add(PersistentState.Id, CompletedDownloadState.Hidden, isHidden);
+        if (tx == null)
+            await transaction.Commit();
+    }
```

Comment:
Why is `tx` nullable and why is this method async? Wouldn't this suffice?

```suggestion
    /// <inheritdoc />
    public void SetIsHidden(bool isHidden, ITransaction tx)
    {
        if (PersistentState.Status != DownloadTaskStatus.Completed) return;
        tx.Add(PersistentState.Id, CompletedDownloadState.Hidden, isHidden);
    }
```

Reply:
Initial implementation allowed setting single tasks to hidden, without caller having to manage the transaction. Now everything uses the multi element version so I can change this